ocean_lib/ocean/ocean_assets.py

```diff
@@ -666,6 +664,111 @@ def pay_for_service(
 
         return tx_id
 
+    @enforce_types
+    def pay_for_compute_service(
+        self,
+        datasets: List[ComputeInput],
+        algorithm_data: Union[ComputeInput, AlgorithmMetadata],
```

Comment:
I'll answer my own question here :facepalm: 

When ordering a compute job using `AlgorithmMetadata` (aka. a "raw algorithm") there are no consume market order fees for the raw algorithm. 

This is because the raw algorithm doesn't exist on-chain as a datatoken and there is no `startOrder` method where consume market order fees are collected.